Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -100,8 +100,8 @@ private static class NativeMethods
             [return: MarshalAs(UnmanagedType.U1)]
             public static extern bool refresh(SharedRealmHandle sharedRealm, out NativeException ex);
 
-            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "shared_realm_get_table", CallingConvention = CallingConvention.Cdecl)]
-            public static extern IntPtr get_table(SharedRealmHandle sharedRealm, [MarshalAs(UnmanagedType.LPWStr)] string tableName, IntPtr tableNameLength, out NativeException ex);
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "shared_realm_get_table_key", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void get_table_key(SharedRealmHandle sharedRealm, [MarshalAs(UnmanagedType.LPWStr)] string tableName, IntPtr tableNameLength, out TableKey tableKey, out NativeException ex);
```

Comment:
Try typedefing `realm_class_key` similar to what the C API does: https://github.com/realm/realm-core/pull/3955/files#diff-ac5ca02b88ba9910d618bac409ad7a1e3b281cdaf9f64840469c21870b7114a1R74

and have it as a return value.